cmd/snap-confine/mount-support.c

```diff
@@ -363,9 +363,19 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		// where $ROOT is either "/" or the "/snap/core/current"
 		// that we are re-execing from
 		char *src = LIBEXECDIR;
-		char self[PATH_MAX + 1] = { 0, };
-		if (readlink("/proc/self/exe", self, sizeof self) > 0) {
+		char self[PATH_MAX] = { 0, };
```

Comment:
Note I didn't mean for you to remove PATH_MAX+1 here. Your change for `sizeof(self) -1` guards against the issue I was referring to (so this code is safe), but now it means it will fail in the unlikely event that a valid path has length of PATH_MAX.